packages/react-components/src/components/VerticalGallery.tsx

```diff
@@ -102,10 +104,21 @@ export const VerticalGallery = (props: VerticalGalleryProps): JSX.Element => {
 
   const numberOfChildren = React.Children.count(children);
   const lastPage = Math.ceil(numberOfChildren / childrenPerPage);
+  const childrenArray = React.Children.toArray(children);
+
+  const indexesArray: number[][] = useMemo(() => {
+    return bucketize([...Array(numberOfChildren).keys()], childrenPerPage);
+  }, [numberOfChildren, childrenPerPage]);
+
+  useEffect(() => {
+    if (setTilesToRender && indexesArray) {
+      setTilesToRender(indexesArray[page - 1]);
+    }
+  }, [indexesArray, setTilesToRender, page]);
 
   const paginatedChildren: React.ReactNode[][] = useMemo(() => {
```

Comment:
same here, generate paginatedChildren from indexesArray instead

Reply:
Logic in different Gallery components look similar - extract these logics into a common util function would be nice - but fully depending on you